File: samples/MinimalZeroInstall.py
import clr
clr.AddReferenceToFile("NanoByte.Common.dll", "ZeroInstall.Services.dll", "ZeroInstall.Store.dll")

import sys
from NanoByte.Common.Tasks import CliTaskHandler
from ZeroInstall.Model import FeedUri, Requirements
from ZeroInstall.Services import ServiceProvider
from ZeroInstall.Services.Feeds import SelectionsManagerExtensions

services = ServiceProvider(CliTaskHandler())
requirements = Requirements(FeedUri(sys.argv[1]))
selections = services.Solver.Solve(requirements)
for implementation in SelectionsManagerExtensions.GetUncachedImplementations(services.SelectionsManager, selections):
    services.Fetcher.Fetch(implementation)
services.Executor.Start(selections)

File: samples/MinimalZeroInstall.fs
open NanoByte.Common.Tasks
open ZeroInstall.Model
open ZeroInstall.Services
open ZeroInstall.Services.Feeds

let services = new ServiceProvider(new CliTaskHandler())
let solve = services.Solver.Solve
let uncached = services.SelectionsManager.GetUncachedImplementations
let fetch = services.Fetcher.Fetch
let execute = services.Executor.Start

let run requirements =
    let selections = solve requirements
    for implementation in (uncached selections) do
        fetch implementation
    execute selections

[<EntryPoint>]
let main args =
    ignore(run(new Requirements(new FeedUri(args.[0]))))
    0
